### PR TITLE
Datatables checkbox

### PR DESCRIPTION
**Pull request by aapomm**
### Summary

The current implementation of the Datatable, is still based on the old behavior of the checkboxes. This doesn’t work with the built-in events that we can hook on when implementing the Issues merge, multi delete, etc.

**Proposed solution**

Implement this: https://datatables.net/extensions/select/examples/initialisation/checkbox.html

### How to test

1. Open the issues page in a project.
2. Confirm that you can select multiple rows.
3. Confirm that the Select all button works as expected.

### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

Thanks for contributing to Dradis!

### Copyright assignment

Collaboration is difficult with commercial closed source but we want
to keep as much of the OSS ethos as possible available to users
who want to fix it themselves.

In order to unambiguously own and sell Dradis Framework commercial
products, we must have the copyright associated with the entire
codebase. Any code you create which is merged must be owned by us.
That's not us trying to be a jerks, that's just the way it works.

Please review the [CONTRIBUTING.md](https://github.com/dradis/dradis-ce/blob/master/CONTRIBUTING.md)
file for the details.

You can delete this section, but the following sentence needs to
remain in the PR's description:

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List

- [ ] Added a CHANGELOG entry

### CODE REVIEW

File: app/assets/javascripts/shared/dradis_datatable.js [hunk 1]
```diff
@@ -2,35 +2,129 @@ class DradisDatatable {
...
+        "<'row'<'col-lg-12'tr>>" +
+        "<'dataTables_footer_content'ip>",
       initComplete: function (settings) {
         settings.oInstance.wrap("<div class='table-wrapper'></div>");
       },
-      pageLength: 25
+      lengthChange: false,
+      pageLength: 25,
+      select: {
+        selector: 'td:first-child',
```
Comment (sean-yeoh) [inline app/assets/javascripts/shared/dradis_datatable.js:60]: Shouldn't this be `td.select-checkbox`?

File: app/assets/javascripts/shared/dradis_datatable.js [hunk 2]
```diff
@@ -2,35 +2,129 @@ class DradisDatatable {
...
+    // Remove default datatable button listener to make the checkbox "checking"
+    // work, before adding our own click handler.
+    $selectAllBtn.off('click.dtb').click( function (){
+      if (that.areAllSelected()) {
+        that.dataTable.rows().deselect();
+      }
+      else {
+        that.dataTable.rows().select();
+      }
+    });
```
Comment (sean-yeoh) [inline app/assets/javascripts/shared/dradis_datatable.js:122]: I'm still kinda confused with this, why is this needed?

Comment (aapomm) [reply app/assets/javascripts/shared/dradis_datatable.js:122]: @sean-yeoh we're not using the Datatables' built-in selectAll button since it doesn't fit our use case. That means we have to implement the selectAll/deselectAll functionality ourselves.

File: app/assets/javascripts/shared/dradis_datatable.js (hunk 1 above)
Comment (aapomm) [reply app/assets/javascripts/shared/dradis_datatable.js]: Both works, but I like the `select-checkbox` more for the specificity.

**Review by etdsoft — APPROVED** [review]
